core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -1595,10 +1594,8 @@ class SparkContext(config: SparkConf) extends Logging with ExecutorAllocationCli
         key = uri.getScheme match {
           // A JAR file which exists only on the driver node
           case null | "file" =>
-            // yarn-standalone is deprecated, but still supported
-            if (SparkHadoopUtil.get.isYarnMode() &&
-                (master == "yarn-standalone" || master == "yarn-cluster")) {
-              // In order for this to work in yarn-cluster mode the user must specify the
+            if (SparkHadoopUtil.get.isYarnMode() && (master == "yarn" && deployMode == "cluster")) {
```

Comment:
not your code, but this is kind of weird. At this point of master is `yarn` we should be pretty sure `isYarnMode` is true.